### ISSUE TITLE
clippy suggested invalid code in from_iter_instead_of_collect

### ISSUE THREAD

**Issue by stevelr** [rust-lang/rust-clippy#7616]
Clippy gave this warning in rule `clippy::from_iter_instead_of_collect` The suggestion to replace is good, however the code replacement recommended is garbage and won't even compile:

```
Warning: usage of `FromIterator::from_iter`
   --> src/util.rs:160:37
    |
160 |             /*'b'|*/ _ => JV::Array(Vec::from_iter(v.into_iter().map(JV::from))),
    |                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: use `.collect()` instead of `::from_iter()`: `v.into_iter().map(JV::from).collect::<Vec::from_iter(v.into_iter().map(JV<_>>()`
    |
    = note: `#[warn(clippy::from_iter_instead_of_collect)]` on by default
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#from_iter_instead_of_collect
```

The correct code replacement should have been
```
     v.into_iter().map(JV::from).collect::<Vec<_>>())
```

**Comment by giraffate** [comment]
Can you provide a minimal reproduction on the playground?

**Comment by stevelr** [comment]
Playground uses clippy 1.56,  even when selecting rust stable 1.54. I cannot reproduce it on playground. 

If I run on my machine with rust stable 1.54 (cargo clippy -V shows 1.54), it reproduces.
If I run nightly (rust 1.56 and cargo clippy -V shows 1.56), it does not reproduce.
It also does not recommend using collect instead of from_iter. So in 1.56,  it's as if the rule itself is disabled or didn't detect the case - i.e., a false-negative for this rule.

Please let me know if you want me to file a separate issue on the false-negative.  I had noticed last week when I filed this that there were concerns about false positives, so I suppose it was throttled down intentionally.

**Comment by giraffate** [comment]
`from_iter_instead_of_collect` was moved to `pedantic` category (allow by default) at 1.55, so you need to add an attribute like `#![warn(clippy::from_iter_instead_of_collect)]` to enable it on nightly.
